pkg/cloud/gcp/assets/validating-admission-policy.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: network-tier-annotation-validation-policy
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups: [""]
+        apiVersions: ["v1"]
+        operations: ["CREATE", "UPDATE"]
+        resources: ["services"]
+  variables:
+    - name: hasNetworkTier
+      expression: has(object.metadata.annotations) && 'cloud.google.com/network-tier' in object.metadata.annotations
+    - name: networkTierValue
+      expression: object.metadata.?annotations['cloud.google.com/network-tier'].orValue("")
+    - name: hasOldNetworkTier
+      expression: has(oldObject) && has(oldObject.metadata) && has(oldObject.metadata.annotations) && 'cloud.google.com/network-tier' in oldObject.metadata.annotations
+    - name: oldNetworkTierValue
+      expression: oldObject.metadata.?annotations['cloud.google.com/network-tier'].orValue("")
+    - name: valueHasChanged
+      expression: |
+        variables.hasOldNetworkTier &&
+        variables.hasNetworkTier &&
+        variables.oldNetworkTierValue != variables.networkTierValue
+  validations:
+    - expression: |
+        !variables.hasNetworkTier || variables.networkTierValue in ['Standard', 'Premium']
+      message: "The annotation 'cloud.google.com/network-tier', if specified, must be either 'Standard' or 'Premium'."
+    - expression: |
+        !variables.hasNetworkTier || !variables.valueHasChanged || variables.networkTierValue in ['Standard', 'Premium']
+      message: "The annotation 'cloud.google.com/network-tier' must not change to an invalid value."
```

Comment:
These validations won't allow ratcheting correctly.

You need to have only one of these two, the latter, but with the message from the previous